redis/runtime/src/main/java/io/quarkiverse/langchain4j/redis/RedisEmbeddingStore.java

```diff
@@ -0,0 +1,246 @@
+package io.quarkiverse.langchain4j.redis;
+
+import static dev.langchain4j.internal.Utils.randomUUID;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.jboss.logging.Logger;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+
+import dev.langchain4j.data.document.Metadata;
+import dev.langchain4j.data.embedding.Embedding;
+import dev.langchain4j.data.segment.TextSegment;
+import dev.langchain4j.internal.Json;
+import dev.langchain4j.store.embedding.EmbeddingMatch;
+import dev.langchain4j.store.embedding.EmbeddingStore;
+import io.quarkiverse.langchain4j.QuarkusJsonCodecFactory;
+import io.quarkiverse.langchain4j.redis.runtime.RedisSchema;
+import io.quarkus.redis.datasource.ReactiveRedisDataSource;
+import io.quarkus.redis.datasource.json.ReactiveTransactionalJsonCommands;
+import io.smallrye.mutiny.Uni;
+import io.vertx.mutiny.redis.client.Command;
+import io.vertx.mutiny.redis.client.Request;
+import io.vertx.mutiny.redis.client.Response;
+
+public class RedisEmbeddingStore implements EmbeddingStore<TextSegment> {
```

Comment:
Mostly a question for @geoand - any possibility to have an async version?
In this case, it's not a big deal, we can use a virtual thread (our Redis client support them).

Reply:
We can yes, but currently there is no async mechanism in langchain4j for embedding stores. We would probably need to work upstream to make that happen or have our types (same goes for other aspects as well, not just embeddings)